File: src/bnl-dark-api/bnl-dark-api/Models/TimeTableEntry.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace bnl_dark_api.Models;

public class TimeTableEntry : ITimeTableEntry
{
    public int Id { get; set; }
    [ForeignKey(nameof(Robot))]
    public int RobotId { get; set; }
    public Robot? Robot { get; set; }
    
    [Required]
    public DateTimeOffset StartTime { get; set; }
    public IEnumerable<TherapyIteration>? TherapyIterations { get; set; }
}

public interface ITimeTableEntry : IId
{
    public int RobotId { get; set; }
    public Robot Robot { get; set; }
    public DateTimeOffset StartTime { get; set; }
    public IEnumerable<TherapyIteration>? TherapyIterations { get; set; }
}

File: src/bnl-dark-web/src/app/robot-management/models/time-table-entry.ts
import { IRobot } from './robot';
import { ITherapyIteration } from './therapy-iteration';

export interface ITimeTableEntry {
  id: number;
  robotId: number;
  robot: IRobot;
  startTime: Date;
  therapyIterations: ITherapyIteration[];
}
